src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs

```diff
@@ -66,19 +66,11 @@ internal class HttpFunctionInvocationDispatcher : IFunctionInvocationDispatcher
         internal Task InitializeHttpWorkerChannelAsync(int attemptCount, CancellationToken cancellationToken = default)
         {
             _httpWorkerChannel = _httpWorkerChannelFactory.Create(_scriptOptions.RootScriptPath, _metricsLogger, attemptCount);
-            _httpWorkerChannel.StartWorkerProcessAsync(cancellationToken).ContinueWith(workerInitTask =>
-                 {
-                     if (workerInitTask.IsCompleted)
-                     {
-                         _logger.LogDebug("Adding http worker channel. workerId:{id}", _httpWorkerChannel.Id);
-                         SetFunctionDispatcherStateToInitializedAndLog();
-                     }
-                     else
-                     {
-                         _logger.LogWarning("Failed to start http worker process. workerId:{id}", _httpWorkerChannel.Id);
-                     }
-                 });
-            return Task.CompletedTask;
+            return _httpWorkerChannel.StartWorkerProcessAsync(cancellationToken).ContinueWith(workerInitTask =>
+            {
+                _logger.LogDebug("Adding http worker channel. workerId:{id}", _httpWorkerChannel.Id);
+                SetFunctionDispatcherStateToInitializedAndLog();
+            }, TaskContinuationOptions.OnlyOnRanToCompletion);
```

Comment:
`TaskContinuationOptions.OnlyOnRanToCompletion` is this enough to handle faulted and cancelled tasks? i.e only runs if prev task is success?

Reply:
Never mind. TaskContinuationOptions.OnlyOnRanToCompletion is the right option!